### PR TITLE
feat: 🎸 pull openserach pass from secretsmanager; rotate weekly

### PR DESCRIPTION
**Pull request by jaskaransarkaria**
* migrates opensearch password from being stored in parameter store (ssm) to secrets manager
* adds a lambda to rotate the password weekly to meet security compliance standards (@kayleigh has details)
* updates opensearch index state management policy (the policy was previously not attached to any indices and so was not deleting logs after 90 days)

### CODE REVIEW

File: modules/opensearch/files/prune_index_policy.json [hunk 1]
```diff
@@ -20,7 +20,7 @@
         "actions": [
           {
             "replica_count": {
-              "number_of_replicas": 0
+              "number_of_replicas": 1
```
Comment (jaskaransarkaria) [inline modules/opensearch/files/prune_index_policy.json:23]: https://opensearch.org/docs/latest/opensearch/index/#primary-and-replica-shards

File: modules/opensearch/data.tf [hunk 2]
```diff
@@ -49,13 +49,50 @@ data "template_file" "snapshot_s3_lambda_policy" {
 
   vars = {
     es_role_arn              = aws_iam_role.snapshot_create.arn
-    es_domain_arn            = aws_elasticsearch_domain.es.arn
+    es_domain_arn            = aws_elasticsearch_domain.os.arn
```
Comment (tomvaughan77) [inline modules/opensearch/data.tf:52]: Nit: Should these vars be renamed to `os_role_arn` and `os_domain_arn`?

**Review by tomvaughan77 — DISMISSED** [review]
💯

File: modules/opensearch/security_groups.tf [hunk 3]
```diff
@@ -49,3 +49,27 @@ resource "aws_security_group_rule" "snapshot_lambda_ingress" {
   security_group_id        = data.aws_security_group.es.id
   source_security_group_id = data.aws_security_group.snapshot_lambda.id
 }
+
+resource "aws_security_group_rule" "allow_secret_rotation_lambda_ingress_from_endpoint" {
+  description = "Allow secrets rotation lambda out to VPCE"
```
Comment (C-gyorfi) [inline modules/opensearch/security_groups.tf:54]:
```suggestion
  description = "Allow secrets rotation lambda into to VPCE"
```

Sorry for nit picking 😅

**Review by C-gyorfi — DISMISSED** [review]
LGTM

**Review by C-gyorfi — APPROVED** [review]

**Pull request merged**